src/core/thread/anycast_locator.cpp

```diff
@@ -0,0 +1,163 @@
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements Anycast Locator functionality.
+ */
+
+#include "anycast_locator.hpp"
+
+#if OPENTHREAD_CONFIG_TMF_ANYCAST_LOCATOR_ENABLE
+
+#include "common/code_utils.hpp"
+#include "common/instance.hpp"
+#include "common/locator_getters.hpp"
+#include "thread/thread_tlvs.hpp"
+#include "thread/uri_paths.hpp"
+
+namespace ot {
+
+AnycastLocator::AnycastLocator(Instance &aInstance)
+    : InstanceLocator(aInstance)
+    , mCallback(nullptr)
+    , mContext(nullptr)
+    , mAnycastLocate(UriPath::kAnycastLocate, HandleAnycastLocate, this)
+
+{
+    Get<Tmf::Agent>().AddResource(mAnycastLocate);
+}
+
+Error AnycastLocator::Locate(const Ip6::Address &aAnycastAddress, Callback aCallback, void *aContext)
+{
+    Error            error   = kErrorNone;
+    Coap::Message *  message = nullptr;
+    Ip6::MessageInfo messageInfo;
+
+    VerifyOrExit((aCallback != nullptr) && Get<Mle::Mle>().IsAnycastLocator(aAnycastAddress),
+                 error = kErrorInvalidArgs);
+
+    message = Get<Tmf::Agent>().NewMessage();
+    VerifyOrExit(message != nullptr, error = kErrorNoBufs);
+
+    SuccessOrExit(error = message->InitAsConfirmablePost(UriPath::kAnycastLocate));
+    SuccessOrExit(error = message->SetPayloadMarker());
+
+    if (mCallback != nullptr)
+    {
+        IgnoreError(Get<Tmf::Agent>().AbortTransaction(HandleResponse, this));
+    }
+
+    messageInfo.SetSockAddr(Get<Mle::MleRouter>().GetMeshLocal16());
+    messageInfo.SetPeerAddr(aAnycastAddress);
+    messageInfo.SetPeerPort(Tmf::kUdpPort);
+
+    SuccessOrExit(error = Get<Tmf::Agent>().SendMessage(*message, messageInfo, HandleResponse, this));
+
+    mCallback = aCallback;
+    mContext  = aContext;
+
+exit:
+    FreeMessageOnError(message, error);
+    return error;
+}
+
+void AnycastLocator::HandleResponse(void *               aContext,
+                                    otMessage *          aMessage,
+                                    const otMessageInfo *aMessageInfo,
+                                    Error                aError)
+{
+    static_cast<AnycastLocator *>(aContext)->HandleResponse(
+        static_cast<Coap::Message *>(aMessage), static_cast<const Ip6::MessageInfo *>(aMessageInfo), aError);
+}
+
+void AnycastLocator::HandleResponse(Coap::Message *aMessage, const Ip6::MessageInfo *aMessageInfo, Error aError)
+{
+    OT_UNUSED_VARIABLE(aMessageInfo);
+
+    uint16_t            rloc16  = Mac::kShortAddrInvalid;
+    const Ip6::Address *address = nullptr;
+    Ip6::Address        meshLocalAddress;
+
+    SuccessOrExit(aError);
+    OT_ASSERT(aMessage != nullptr);
+
+    meshLocalAddress.SetPrefix(Get<Mle::Mle>().GetMeshLocalPrefix());
+    SuccessOrExit(Tlv::Find<ThreadMeshLocalEidTlv>(*aMessage, meshLocalAddress.GetIid()));
+    SuccessOrExit(Tlv::Find<ThreadRloc16Tlv>(*aMessage, rloc16));
+
+#if OPENTHREAD_FTD
+    if (Get<AddressResolver>().UpdateCacheEntry(meshLocalAddress, rloc16) != kErrorNone)
```

Comment:
Adding a note here to remember to update this when/if PR https://github.com/openthread/openthread/pull/6604 is merged.

Reply:
Merged :)

Reply:
Thanks. Just rebased and updated it